File: 00_interview_question/08_reverse_string/main.c
#include <stdio.h>
#include <string.h>

void reverse_string( char *str) {
    int length = strlen(str);
    char* start = str;
    char* end = str + length -1;
    while(start < end) {
        char temp = *start;
        *start = *end;
        *end = temp;

        start++;
        end--;
    }
}

int main() {
    char str[] = "abcd";
    printf("Original string: %s\n", str);
    
    reverse_string(str);
    
    printf("Reversed string: %s\n", str);
    
    return 0;
}

File: 00_interview_question/08_reverse_string/main.py
def reverse(s):
    s_list = list(s)
    s_list[0], s_list[-1] = s_list[-1], s_list[0]
    return''.join(s_list)

def main():
    s = "abcd"
    print("Original string:", s)
    
    reversed_str = reverse(s)
    
    print("Reversed string:", reversed_str)

if __name__ == "__main__":
    main()
